lib/Pipedrive.js

```diff
@@ -28,6 +30,77 @@
 			callback(error, collectionItems, additionalData);
 		}, false);
 	};
+	exports.OAuthAuthenticate = function (client_id, redirect_uri, res) {
+	    res.redirect('https://oauth.pipedrive.com/oauth/authorize?' +
+            qs.stringify({
+                client_id: client_id,
+                redirect_uri: redirect_uri,
+            }))
+	};
+
+	exports.OAuthAuthenticateCallback = function (req, res, redirect_uri, clientId, clientSecret) {
+	    var code = req.query.code || null;
+	    var authOptions = {
+	        url: 'https://oauth.pipedrive.com/oauth/token',
+            form: {
+	            code: code,
+                redirect_uri: redirect_uri,
+                grant_type: 'authorization_code'
+            },
+			headers: {
+				'Authorization': getBasicAuthHeader(clientId, clientSecret)
+			},
+            json: false
+	    };
+	    return new Promise(function (resolve) {
+			var req = rest.post(authOptions.url,
+				{
+					json: authOptions.json,
+					multipart: false,
+					data: authOptions.form,
+					basic_auth: authOptions.headers['Authorization']
+				});
+
+			req.on('complete', function(data) {
+				var body = JSON.parse(data);
```

Comment:
Can we have `try..catch` in case if JSON isn't valid? Otherwise we will have unhandled promise rejection.